# HG changeset patch
# User coleenp
# Date 1299199896 18000
# Node ID 9f44e9aad2d9528c6a7c59f29be7974685d6b629
# Parent  a3c0ec0428a2564a70a33a0348553e9d4c994c63
7022999: Can't build with FORCE_TIERED=0
Summary: Put UseFastLocking test under #ifdef COMPILER1
Reviewed-by: kvn, phh, never, dcubed

diff --git a/src/share/vm/runtime/arguments.cpp b/src/share/vm/runtime/arguments.cpp
--- openjdk/hotspot/src/share/vm/runtime/arguments.cpp
+++ openjdk/hotspot/src/share/vm/runtime/arguments.cpp
@@ -3114,7 +3114,11 @@
   // Turn off biased locking for locking debug mode flags,
   // which are subtlely different from each other but neither works with
   // biased locking.
-  if (!UseFastLocking || UseHeavyMonitors) {
+  if (UseHeavyMonitors
+#ifdef COMPILER1
+      || !UseFastLocking
+#endif // COMPILER1
+    ) {
     if (!FLAG_IS_DEFAULT(UseBiasedLocking) && UseBiasedLocking) {
       // flag set to true on command line; warn the user that they
       // can't enable biased locking here
